content/blog/wallet-descriptors.md

```diff
@@ -0,0 +1,164 @@
++++
+title = "Wallet Descriptors"
+date = 2022-06-01
+description = "Wallet descriptors or output script descriptor are plain-text language that helps wallet move from public/private keys to scripts and addresses"
+[extra]
+    image = "https://res.cloudinary.com/vladimirfomene/image/upload/c_scale,w_1039/v1653223416/blog/malleability.jpg"
+    url = "https://www.vladimirfomene.com/blog/wallet-descriptors/"
++++
+
+Welcome back to the blog! This week we are going to be looking at the issues that led to the creation of wallet descriptors, look at their syntax and talk about their implications for wallets.
+
+## Introduction
+
+To the everyday Bitcoin user, wallets seem to be working fairly well and can easily be backed up thanks to BIP39 mnemonic phrases. Though the tree of keys wallet or the wallet generated from the mnemonic phrase often called the legacy wallet works well, it has a couple of shortcomings. Due to these shortcomings, BIP380 was proposed to resolve some of the issues faced by
+the current wallet architecture.
+
+## Why  Wallet Descriptors
+
+As mentioned in the introduction, with the old wallet we only have keys and it is not clear to the 
+wallet software which type of address types (P2PKH, P2WPKH, P2WSH, etc) to generate from these keys.
+As a consequence, wallet restoration is very computationally intensive, for each key, wallets have to scan the whole chain for different ScriptPubKey types to calculate a user's balance.
+
+Although, BIP44, BIP49 and BIP84 provide standard derivation paths to be used by all wallets, not every wallet supports these BIPs. As a direct consequence, some wallets will not be able to
+derive a user's balance given a BIP39 mnemonic phrase as they might be using a different derivation path from the wallet that produced the backup. This is will create an illusion of a loss of funds for a user thereby providing a very poor user experience.
+
+Last but not the least, this current architecture also doesn't provide the mechanism for wallets to easily watch funds in a multisig. 
```

Comment:
...and perhaps most importantly to be able to monitor and sign transactions related to arbitrary scripts, without specific logic being added to every single wallet implementation